File: app/src/main/java/net/chris/demo/guardian/ui/model/ConcreteCommand.java
package net.chris.demo.guardian.ui.model;

public class ConcreteCommand extends Command {

    private CommandReceiver receiver;

    private ConcreteCommand(CommandReceiver receiver) {
        this.receiver = receiver;
    }

    @Override
    public void execute() {
        if (receiver == null) {
            return;
        }
        receiver.handleCommand();
    }

    public static Command create(CommandReceiver receiver) {
        return new ConcreteCommand(receiver);
    }

}

File: app/src/main/kotlin/net/chris/demo/guardian/ui/model/ConcreteCommand.kt
package net.chris.demo.guardian.ui.model

class ConcreteCommand private constructor(private val receiver: CommandReceiver?) : Command() {

    override fun execute() {
        if (receiver == null) {
            return
        }
        receiver.handleCommand()
    }

    companion object {

        fun create(receiver: CommandReceiver?): Command {
            return ConcreteCommand(receiver)
        }
    }

}
